src/DataTypes/Serializations/SerializationArray.cpp

```diff
@@ -614,29 +615,124 @@ void SerializationArray::serializeTextJSONPretty(const IColumn & column, size_t
     writeChar(']', ostr);
 }
 
-
-void SerializationArray::deserializeTextJSON(IColumn & column, ReadBuffer & istr, const FormatSettings & settings) const
+namespace
 {
-    deserializeTextImpl(column, istr,
-        [&](IColumn & nested_column)
+template <typename ReturnType>
+ReturnType deserializeTextJSONArrayImpl(IColumn & column, ReadBuffer & istr, const SerializationPtr & nested, const FormatSettings & settings)
```

Comment:
Yes, that's what I was planning to do when you started reviewing my code. I decided against refactoring the code I added in `JSONUtils::readField` since I felt it was different enough from the rest.